lib/engine/game/g_18_usa/step/scrap_train.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+require_relative 'scrap_train_module'
+module Engine
+  module Game
+    module G18USA
+      module Step
+        class ScrapTrain < Engine::Step::Base
+          include ScrapTrainModule
+          def actions(entity)
+            actions = []
+            actions << 'scrap_train' if entity == current_entity && entity.corporation? &&
```

Comment:
call can_scrap_train? here instead of duplicating the logic.

Reply:
I've consolidated this logic into the scrap_train module and I was able to consolidate the `scrap_train_by_<foo>` methods there because <foo>.owned_by?(<foo>) is true if it's a corporaton